### PR TITLE
use simulated/fake components

### PR DESCRIPTION
**Pull request by souravran**

### CODE REVIEW

File: cob_bringup/drivers/sick_flexisoft.launch [hunk 1]
```diff
@@ -14,6 +15,7 @@
 		<remap from="diagnostics" to="/diagnostics"/>
 	</node>
 
-
+	<!-- FIXME remove relayboard_sim with a cob_safety_controller_sim which makes use of simulated safety fields from laser -->
```
Comment (fmessmer) [inline cob_bringup/drivers/sick_flexisoft.launch:18]: added to https://github.com/ipa320/msh/issues/586#issuecomment-275435124

**Review by floweisshardt — APPROVED** [review]

File: cob_controller_configuration_gazebo/launch/robots/default_controllers_cob4-2.launch [hunk 2]
```diff
@@ -6,19 +6,12 @@
...
-	<include file="$(find cob_bringup)/drivers/relayboard.launch">
-		<arg name="sim" value="true"/>
-	</include>
+	<node ns="sensorring" pkg="cob_controller_configuration_gazebo" type="gazebo_services.py" name="gazebo_services" cwd="node" respawn="false" output="screen" />
+	<node ns="head" pkg="cob_controller_configuration_gazebo" type="gazebo_services.py" name="gazebo_services" cwd="node" respawn="false" output="screen" />
 
 	<node name="image_proc" pkg="image_proc" type="image_proc" ns="head_cam"/>
 
-	<node name="fake_monitor" pkg="cob_controller_configuration_gazebo" type="fake_monitoring.py" args="--diag-hostnames='light, safety, base_laser_front, base_laser_right, base_laser_left, -b1, -t1, -t2, -t3, -s1, -h32, torso_cam3d_right, torso_cam3d_left, head_cam, sensorring_cam3d, torso/driver, torso_cam3d_down, base/driver, joystick'"/>
+	<!-- FIXME: remove all this and use arg:=sim within the component launch files to distinguish between real component or sim/fake component -->
```
Comment (fmessmer) [inline cob_controller_configuration_gazebo/launch/robots/default_controllers_cob4-2.launch:14]: added to https://github.com/ipa320/msh/issues/586#issuecomment-275435124

**Comment by fmessmer** [comment]
rebase/merge as #605 is merged - this will fix travis

File: cob_bringup/robots/cob4-10.xml [hunk 3]
```diff
@@ -35,15 +35,15 @@
...
-			</include>
 		</group>
 
+		<include file="$(find cob_bringup)/drivers/sick_flexisoft.launch">
+			<arg name="robot" value="$(arg robot)"/>
+			<arg name="sim" value="$(arg sim)"/>
+		</include>
+		<include file="$(find cob_bringup)/drivers/bms.launch">
+			<arg name="robot" value="$(arg robot)"/>
+		</include>
```
Comment (fmessmer) [inline cob_bringup/robots/cob4-10.xml:46]: can `bms.launch` really be run in simulation without any adjustment?

Comment (floweisshardt) [reply cob_bringup/robots/cob4-10.xml:46]: no. Even if the node does not crash there is for sure no outcome. We need to use a fake component if `sim:=true`

Comment (fmessmer) [reply cob_bringup/robots/cob4-10.xml:46]: at the moment there is no such distinction between sim/hw in bms.launch
@souravran do you already have such a fake_bms script?

Comment (souravran) [reply cob_bringup/robots/cob4-10.xml:46]: I dont have a separate fake_bms script but to go on with the simulation, I just publish to the power_state.

Comment (fmessmer) [reply cob_bringup/robots/cob4-10.xml:46]: Like this?
`<node pkg="rostopic" type="rostopic" args="pub <topic_name> <topic_type> <message_content> -r<rate>" name="<name>" output="screen"/>`
Or how do you publish power_state?

Could you put that into the bms.launch then...for now?
Together with a comment in https://github.com/ipa320/msh/issues/586

Comment (souravran) [reply cob_bringup/robots/cob4-10.xml:46]: Yes I will do that.

**Pull request merged**